### PR TITLE
Emit armv8.4 ldapur* for volatile loads with contained offsets

### PR DESCRIPTION
**Pull request by EgorBo**
Contributes to https://github.com/dotnet/runtime/issues/64457

```cs
volatile int a;
volatile int b;
volatile int c;
volatile int d;

[Benchmark]
public int Test()
{
    return a + b + c + d;
}
```
Current codegen for `Test`:
```asm
; Assembly listing for method P:Test():int:this (Tier1)		
            stp     fp, lr, [sp, #-0x10]!		
            mov     fp, sp		
            add     x1, x0, #8		
            ldapr   w1, [x1]		
            add     x2, x0, #12		
            ldapr   w2, [x2]		
            add     w1, w1, w2		
            add     x2, x0, #16		
            ldapr   w2, [x2]		
            add     w1, w1, w2		
            add     x0, x0, #20		
            ldapr   w0, [x0]		
            add     w0, w1, w0		
            ldp     fp, lr, [sp], #0x10		
            ret     lr		
; Total bytes of code 60
```
New codegen for `Test`:
```asm
; Assembly listing for method P:Test():int:this (Tier1)
            stp     fp, lr, [sp, #-0x10]!
            mov     fp, sp
            ldapur  w1, [x0, #0x08]
            ldapur  w2, [x0, #0x0C]
            add     w1, w1, w2
            ldapur  w2, [x0, #0x10]
            add     w1, w1, w2
            ldapur  w0, [x0, #0x14]
            add     w0, w1, w0
            ldp     fp, lr, [sp], #0x10
            ret     lr
; Total bytes of code 44
```

Apple M2 Max:

| Method |        Job |               Toolchain |      Mean |     Error |    StdDev | Ratio |
|------- |----------- |------------------------ |----------:|----------:|----------:|------:|
|   Test | Job-LOXHRO |      /Core_Root_PR/corerun | 0.4724 ns | 0.0050 ns | 0.0045 ns |  0.89 |
|   Test | Job-MPJUHL | /Core_Root_main/corerun | 0.5327 ns | 0.0063 ns | 0.0059 ns |  1.00 |

Perf difference is not super exciting but it is what it is 🙂

### CODE REVIEW

File: src/coreclr/jit/emit.h [hunk 1]
```diff
@@ -628,8 +628,8 @@ class emitter
 #define MAX_ENCODED_SIZE 15
 #elif defined(TARGET_ARM64)
 #define INSTR_ENCODED_SIZE 4
-        static_assert_no_msg(INS_count <= 512);
-        instruction _idIns : 9;
+        static_assert_no_msg(INS_count <= 1024);
+        instruction _idIns : 10;
```
Comment (EgorBo) [inline src/coreclr/jit/emit.h:632]: I added a bit here since we now have more than 512 instructions (at that for sure will be increasing in future) but I also removed an used (on arm) `_idCallRegPtr` bit so the total size of the struct remains the same - 16b

File: src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
Comment (neon-sunset) [inline src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt:1]: Is `rcpc2` intentionally not defined for `instructionsetgroup  ,armv8.4-a` at L207?

Comment (EgorBo) [reply src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt:1]: Good catch! fixed

**Comment by EgorBo** [comment]
@TIHan @dotnet/jit-contrib PTAL, simple change, emits `ldapur` where we previously emitted `ldapr+add` because `ldapr` doesn't support IMM offsets

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -8813,8 +8837,6 @@ void emitter::emitIns_Call(EmitCallType          callType,
     {
         /* This is an indirect call (either a virtual call or func ptr call) */
 
-        id->idSetIsCallRegPtr();
```
Comment (TIHan) [inline src/coreclr/jit/emitarm64.cpp:8816]: Why was this removed for arm32?

**Review by TIHan — APPROVED** [review]
LGTM, only one comment on the idSetIsCallingRegPtr.

File: src/coreclr/jit/emitarm64.cpp (hunk 2 above)
Comment (EgorBo) [reply src/coreclr/jit/emitarm64.cpp:8816]: I mentioned it in https://github.com/dotnet/runtime/pull/89681/files#r1278410873 comment, basically, it sets a bit that is unused for arm, so I removed that bit because I needed to extend `_idIns`'s width
